src/net/sourceforge/kolmafia/request/PingRequest.java

```diff
@@ -0,0 +1,59 @@
+package net.sourceforge.kolmafia.request;
+
+import java.util.GregorianCalendar;
+
+public class PingRequest extends GenericRequest {
+
+  private long startTime = 0L;
+  private long endTime = 0L;
+
+  public PingRequest() {
+    this("main.php");
+  }
+
+  public PingRequest(String pingURL) {
+    super(pingURL);
+  }
+
+  @Override
+  public void run() {
+    // You can reuse this request.
+    this.startTime = this.endTime = 0;
+
+    this.startTime = new GregorianCalendar().getTimeInMillis();
```

Comment:
`System.currentTimeMillis()` would avoid the (small) overhead of creating an object, dealing with calendar stuff.

Reply:
or, for estimating time differences, `System.nanoTime()`?

Reply:
This is a good point. I like storing the history in milliseconds, which are a lot easier for the user to read and compare than nanoseconds. I'm not sure we need a million times the precision, although we could certainly calculate with that and just divide by a million when saving the data.